### PR TITLE
Add access to CLOCK_MONOTONIC with nsecs(monotonic)

### PR DESCRIPTION
**Pull request by ajor**
This lets users explicitly request CLOCK_MONOTONIC when it is desired to get a timestamp exclusive of suspend time.

Includes some refactoring of the semantic analyser `test()` overloads and deleted some redundant and out-of-date sections of the reference guide.

### CODE REVIEW

**Review by viktormalik — APPROVED** [review]
Nice cleanup!

**Review by xh4n3 — COMMENTED** [review]
A couple of nits...
(I forgot the submit this review, maybe it is too late now...)

File: src/ast/irbuilderbpf.h [hunk 1]
```diff
@@ -159,7 +159,7 @@ class IRBuilderBPF : public IRBuilder<>
                                const bool inverse,
                                const location &loc,
                                MDNode *metadata);
-  CallInst *CreateGetNs(bool boot_time, const location &loc);
+  CallInst *CreateGetNs(TimestampMode ts, const location &loc);
   CallInst *CreateGetTaiNs(const location &loc);
```
[the remark this answers is not in the document]
Comment (ajor) [reply src/ast/irbuilderbpf.h:163]: Oops, sloppy! Thank you

File: src/ast/irbuilderbpf.cpp [hunk 2]
```diff
@@ -971,23 +971,34 @@ Value *IRBuilderBPF::CreateStrcontains(Value *val1,
...
+  libbpf::bpf_func_id fn;
+  switch (ts)
+  {
+    case TimestampMode::monotonic:
+      fn = libbpf::BPF_FUNC_ktime_get_ns;
+      break;
+    case TimestampMode::boot:
+      fn = bpftrace_.feature_->has_helper_ktime_get_boot_ns()
+               ? libbpf::BPF_FUNC_ktime_get_boot_ns
+               : libbpf::BPF_FUNC_ktime_get_ns;
```
[the remark this answers is not in the document]
Comment (ajor) [reply src/ast/irbuilderbpf.cpp:985]: Yeah we might want to change some stuff around `nsecs()`/`nsecs(boot)`, but I'm not sure exactly in what way and wanted to keep this PR simple. We can do this later if any good ideas come up

**Pull request merged**